allennlp/data/samplers/bucket_batch_sampler.py

```diff
@@ -66,20 +70,24 @@ class BucketBatchSampler(BatchSampler):
     def __init__(
         self,
         data_source: data.Dataset,
-        batch_size: int,
+        batch_size: Optional[int] = None,
+        max_tokens: Optional[int] = None,
         sorting_keys: List[str] = None,
         padding_noise: float = 0.1,
         drop_last: bool = False,
     ):
-
+        assert (batch_size is None) != (max_tokens is None), (
+            "Need to provide exactly one of" "batch_size or max_tokens"
+        )
```

Comment:
Something happened here with the formatting.

Reply:
`black .` did this for me. `black .` has been doing some odd formatting changes for me recently. For instance, this:

```
            m = training_util.get_valid_tokens_mask(reference_seq, self._exclude_indices).sum().item()
            n = training_util.get_valid_tokens_mask(predicted_seq, self._exclude_indices).sum().item()
```

turned into this

```
            m = (
                training_util.get_valid_tokens_mask(reference_seq, self._exclude_indices)
                .sum()
                .item()
            )
            n = (
                training_util.get_valid_tokens_mask(predicted_seq, self._exclude_indices)
                .sum()
                .item()
            )
```